assembl/static/js/app/views/messageColumnsPanel.js

```diff
@@ -0,0 +1,451 @@
+  },
+
+  regions: {
+    messageColumnsList: '@ui.messageColumnsList',
+    ideaAnnouncement: '@ui.ideaAnnouncement'
+  },
+
+  initialize: function(options) {
+    AssemblPanel.prototype.initialize.apply(this, arguments);
+    var that = this,
+        current_idea = this.getGroupState().get('currentIdea'),
+        collectionManager = new CollectionManager();
+    if(this.isViewDestroyed()) {
+      return;
+    }
+    this.translationDataPromise = collectionManager.getUserLanguagePreferencesPromise(Ctx);
+    this.setCurrentIdea(current_idea);
+    this.listenTo(this.getGroupState(), "change:currentIdea", function(groupState) {
+      that.setCurrentIdea(groupState.get('currentIdea'));
+      that.attachmentCollection = that.currentIdea.get('attachments');
+      that.render();
+    });
+    this.listenTo(Assembl.vent, 'messageList:showMessageById', function(id, callback) {
+      that.showMessageById(id, callback);
+    });
+
+    if (current_idea != null) {
+      this.attachmentCollection = current_idea.get('attachments');
+    }
+    // this.listenTo(this.attachmentCollection, 'add remove change', function(){
+    //   console.log("Listening to attachment collection");
+    //   that.render();
+    // });
+
+  },
+  setCurrentIdea: function(idea) {
+    if (this.isViewDestroyed()) {
+      return;
+    }
+    if (idea == null) {
+      idea = this.currentIdea || this.getGroupState().get("currentIdea");
+    }
+    if (this.currentIdea === idea) {
+      return;
+    }
+    if (idea != null) {
+      this.announcementPromise = idea.getApplicableAnnouncementPromise();
+    }
+    this.currentIdea = idea;
+  },
+
+  onRender: function() {
+    if (this.isViewDestroyed()) {
+      return;
+    }
+    var that = this,
+        idea = this.currentIdea;
+    if (idea == undefined) {
+      // after message send, somehow...
+      idea = this.getGroupState().get("currentIdea");
+      this.setCurrentIdea(idea);
+    }
+    if (idea == undefined) {
+      console.warn("WHY is the idea undefined?");
+      return;
+    }
+    var columns = idea.get("message_columns");
+    if (columns === undefined || columns.length === 0) {
+      console.log("TODO: this view should not be alive.");
+      return;
+    }
+    // first approximation
+    // this.ui.ideaColumnHeader.html(idea.get("shortTitle"));
+    this.announcementPromise.then(function(announcement) {
+      if (that.isViewDestroyed() || announcement === undefined) {
+        return;
+      }
+      var announcementMessageView = new Announcements.AnnouncementMessageView({model: announcement});
+      that.showChildView('ideaAnnouncement', announcementMessageView);
+      that.ui.ideaAnnouncement.removeClass('hidden');
+      var attachmentModel = that.attachmentCollection.getSingleAttachment();
+      if (attachmentModel){
+        var announcementImgBackgroundLink = attachmentModel.get('external_url');
+        $('.js_ideaAnnouncement').addClass('background-annoucement');
```

Comment:
Is there really any difference between `this.$()` and `$()`? As I understand it, Backbone just references the global `$()` anyway.